### ISSUE TITLE
Local printers don't display their location

### ISSUE THREAD

**Issue by kylemacfarlane** [google/cups-connector#224]
Cloud printers display their location from CUPS whereas local printers do not.

![connector](https://cloud.githubusercontent.com/assets/99594/15367302/5b358ee8-1d20-11e6-95a8-38cab0a43ea4.png)

**Comment by kylemacfarlane** [comment]
On a similar note, if you pause a printer in CUPS the GCP control panel will extract the state and state-reasons from the tags and show a nice error message. However since the connector still says the printer is online you can still send jobs to the printer as if it's working fine.

Is that how things are meant to work?

**Comment by agoode** [comment]
The whole printer location stuff is a bit magic. See
https://github.com/google/cups-connector/blob/master/gcp/gcp.go#L38

```
	// This prefix tickles a magic spell in GCP so that, for example,
	// the GCP UI shows location as the string found in the
	// printer-location CUPS attribute.
	gcpTagPrefix = "__cp__"
```

Chromium also just reads the magic ``__cp__printer-location`:
https://chromium.googlesource.com/chromium/src/+/599ffa1d8087230b83f5b097cba7a06fc6963f77/chrome/browser/resources/print_preview/data/destination.js#177

I don't see anything about location in the privet protocol:
https://developers.google.com/cloud-print/docs/privet#42-privetinfo-api

It looks like tags are not in CDD, but outside as a parameter to /register:
https://developers.google.com/cloud-print/docs/proxyinterfaces#parameters_7

So either CDD or privet needs to get a proper location field. It looks like GCP needs one too, instead of this magic string tag.

**Comment by kylemacfarlane** [comment]
That is basically what I expected. Some parts of the GCP ecosystem seem to extract CUPS tags (location, state, reason, etc) while others do not. It creates an inconsistent experience.

**Comment by agoode** [comment]
Are there other missing fields between local vs. cloud? You mention state and reason, where are those inconsistently presented?

**Comment by kylemacfarlane** [comment]
It's not a local vs cloud thing but a control panel vs (local and cloud).

For example pause a printer in CUPS and you will get the following at google.com/cloudprint/.

![cups-connector](https://cloud.githubusercontent.com/assets/99594/15734825/486b6c26-288b-11e6-9872-4f927b37e089.png)

But both cloud and local printers will still appear as accepting jobs in Chrome.

Also if you don't pause a printer in CUPS but rather set it to reject jobs then GCP won't detect that anything is wrong with the printer at all. This seems to be because rejecting jobs doesn't affect the printer state in CUPS so another field probably has to be checked.

**Comment by agoode** [comment]
There are 2 problems here. One is the cups-connector not checking the rejection state. I'll file an issue here for that.

The other is a Chrome issue. Can you file a bug at crbug.com/new ?

**Comment by agoode** [comment]
I am trying to figure out if I can fix the privet protocol in some way to allow for location. I'll update this issue if I make progress.

**Comment by kylemacfarlane** [comment]
For reference I opened https://bugs.chromium.org/p/chromium/issues/detail?id=619428

**Comment by agoode** [comment]
I don't think this is going to be fixed soon, but I hope it will be eventually.
